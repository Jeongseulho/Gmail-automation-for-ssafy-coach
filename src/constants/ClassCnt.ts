export const CLASS_CNT = ['2', '3', '4', '5', '6', '7', '8'];
