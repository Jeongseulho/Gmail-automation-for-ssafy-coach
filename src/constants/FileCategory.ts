export const FILE_CATEGORY = {
  DAILY_GATHER: '취합',
  DAILY: '일일',
  JIRA: '주간Jira현황',
  UNIT: '단위업무보고서',
  DAY_OFF: '휴가신청서',
} as const;
