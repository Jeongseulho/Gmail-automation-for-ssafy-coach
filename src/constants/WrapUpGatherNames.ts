export const WRAP_UP_GATHER_NAMES = ['정슬호', '전찬혁', '박단비', '류지윤', '서은진', '김주아', '서정빈', '송소연', '정소정', '최준서', '(이름 직접 입력)'];
