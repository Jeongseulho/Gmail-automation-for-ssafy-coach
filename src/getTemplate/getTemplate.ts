import { getDailyGatherTemplate } from '../utils/dailyGather/getDailyGatherTemplate';
import { getDailyTemplate } from '../utils/daily/getDailyTemplate';
import { parseDaily } from '../utils/daily/parseDaily';

export const getTemplate = (attachedFileName: string) => {
  if (attachedFileName.includes('일일')) {
    if (attachedFileName.includes('취합')) {
      const { date, campus, name } = parseDaily(attachedFileName, true);
      return getDailyGatherTemplate(date, campus, name);
    }
    const { date, campus, classGroup, name } = parseDaily(attachedFileName);
    return getDailyTemplate(date, campus, classGroup || 'x반', name);
  }

  // if (attachedFileName.includes('주간Jira현황')) {
  //   const { date, campus, classGroup, week, name1, name2 } = parseJira(attachedFileName);
  // }
};
