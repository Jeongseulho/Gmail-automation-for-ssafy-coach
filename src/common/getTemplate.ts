import { getDailyGatherTemplate } from '@/utils/dailyGather/getDailyGatherTemplate';
import { FILE_CATEGORY } from '@/constants/FileCategory';
import { getDailyTemplate } from '@/utils/daily/getDailyTemplate';
import { parseDaily } from '@/utils/daily/parseDaily';
import { FileCategoryValue } from '@/types/FileCategoryValue';
import { parseJira } from '@/utils/jira/parseJira';
import { getJiraTemplate } from '@/utils/jira/getJiraTemplate';
import { nameSelect } from './nameSelect';

export const getTemplate = async (attachedFileName: string, fileCategory: FileCategoryValue) => {
  switch (fileCategory) {
    case FILE_CATEGORY.DAILY_GATHER: {
      const { date, campus, name } = parseDaily(attachedFileName, true);
      return getDailyGatherTemplate(date, campus, name);
    }
    case FILE_CATEGORY.DAILY: {
      const { date, campus, classGroup, name } = parseDaily(attachedFileName);
      return getDailyTemplate(date, campus, classGroup || 'x반', name);
    }
    case FILE_CATEGORY.JIRA: {
      const { date, project, campus, classGroup, week, name1, name2 } = parseJira(attachedFileName);
      const selectedName = await nameSelect(name1, name2);
      return getJiraTemplate(date, project, campus, classGroup, week, selectedName);
    }
    default: {
      alert('파일 이름을 파싱, 템플릿을 가져오는 과정에서 에러가 발생했습니다.');
      throw new Error('파일 이름을 파싱, 템플릿을 가져오는 과정에서 에러가 발생했습니다.');
    }
  }
};
