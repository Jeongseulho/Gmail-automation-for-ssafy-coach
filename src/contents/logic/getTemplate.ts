import { FILE_CATEGORY } from '@/constants/FileCategory';
import { FileCategoryValue } from '@/types/FileCategoryValue';
import { nameSelect } from '@/contents/logic/common/nameSelect';
import { parseDaily } from '@/contents/logic/daily/parseDaily';
import { parseJira } from '@/contents/logic/jira/parseJira';
import { parseDayOff } from '@/contents/logic/dayOff/parseDayOff';
import { getDayOffTemplate } from '@/contents/logic/dayOff/getDayOffTemplate';
import { getDailyGatherTemplate } from '@/contents/logic/dailyGather/getDailyGatherTemplate';
import { getDailyTemplate } from '@/contents/logic/daily/getDailyTemplate';
import { getJiraTemplate } from '@/contents/logic/jira/getJiraTemplate';
import { dayOffSelect } from '@/contents/logic/dayOff/dayOffSelect';
import { getUnitWithJiraTemplate } from '@/contents/logic/unitWithJira/getUnitWithJiraTemplate';
import { getUnitTemplate } from './unit/getUnitTemplate';
import { parseUnit } from './unit/parseUnit';
import { weekSelect } from './common/weekSelect';
import { parseWrapUpGather } from './wrapUpGather/parseWrapUpGather';
import { getWrapUpGatherTemplate } from './wrapUpGather/getWrapUpGatherTemplate';
import { parseWrapUp } from './wrapUp/parseWrapUp';
import { getWrapUpTemplate } from './wrapUp/getWrapUpTemplate';
import { WRAP_UP_GATHER_NAMES } from '@/constants/WrapUpGatherNames';
import { CLASS_CNT } from '@/constants/ClassCnt';
import { classCntSelect } from './common/classCntSelect';

export const getTemplate = async (attachedFileName: string, fileCategory: FileCategoryValue, isUnitWithJira: boolean) => {
  if (isUnitWithJira) {
    const { date, project, campus, classGroup, week, name1, name2 } = parseJira(attachedFileName);
    const name = await nameSelect(name1, name2);
    return getUnitWithJiraTemplate(date, project, campus, classGroup, week, name);
  }
  switch (fileCategory) {
    case FILE_CATEGORY.DAILY_GATHER: {
      const { cohort, date, campus, name } = parseDaily(attachedFileName, true);
      const classCnt = await classCntSelect(...CLASS_CNT);
      return getDailyGatherTemplate(cohort, date, campus, name, classCnt);
    }
    case FILE_CATEGORY.DAILY: {
      const { cohort, date, campus, classGroup, name } = parseDaily(attachedFileName);
      return getDailyTemplate(cohort, date, campus, classGroup || 'x반', name);
    }
    case FILE_CATEGORY.JIRA: {
      const { cohort, date, project, campus, classGroup, week, name1, name2 } = parseJira(attachedFileName);
      const name = await nameSelect(name1, name2);
      return getJiraTemplate(cohort, date, project, campus, classGroup, week, name);
    }
    case FILE_CATEGORY.UNIT: {
      const { cohort, date, project, campus, classGroup, name1, name2 } = parseUnit(attachedFileName);
      const name = await nameSelect(name1, name2);
      const selectedWeek = await weekSelect(project);
      return getUnitTemplate(cohort, date, project, campus, classGroup, selectedWeek, name);
    }
    case FILE_CATEGORY.DAY_OFF: {
      const { cohort, date, campus, name } = parseDayOff(attachedFileName);
      const dayOffCategory = await dayOffSelect();
      return getDayOffTemplate(cohort, date, campus, name, dayOffCategory);
    }
    case FILE_CATEGORY.WRAP_UP_GATHER: {
      const { cohort, date, day, campus } = parseWrapUpGather(attachedFileName);
      const selectedName = await nameSelect(...WRAP_UP_GATHER_NAMES);
      return getWrapUpGatherTemplate(cohort, date, day, campus, selectedName);
    }

    case FILE_CATEGORY.WRAP_UP: {
      const { cohort, date, day, campus, name } = parseWrapUp(attachedFileName);
      return getWrapUpTemplate(cohort, date, day, campus, name);
    }

    default: {
      alert('파일 이름을 파싱, 템플릿을 가져오는 과정에서 에러가 발생했습니다(파일 이름이 올바른지 확인해주세요).');
      throw new Error('파일 이름을 파싱, 템플릿을 가져오는 과정에서 에러가 발생했습니다(파일 이름이 올바른지 확인해주세요).');
    }
  }
};
