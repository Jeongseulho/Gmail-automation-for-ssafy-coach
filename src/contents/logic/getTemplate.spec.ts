import { getTemplate } from './getTemplate';
import { FILE_CATEGORY } from '@/constants/FileCategory';

describe('getTemplate 함수', () => {
  it('일일 보고서 템플릿 반환', async () => {
    const template = await getTemplate('10기_240206_일일 프로젝트 진행_공통_구미_1반_정슬호_오프라인.docx', FILE_CATEGORY.DAILY, false);
    expect(template).toMatchSnapshot();
  });
});
