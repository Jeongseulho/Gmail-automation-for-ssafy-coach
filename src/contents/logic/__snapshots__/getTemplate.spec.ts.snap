// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTemplate 함수 일일 보고서 템플릿 반환 1`] = `
{
  "content": "
    <div>안녕하십니까, 10기 구미 실습코치 정슬호입니다.</div>
    <br>
    <div>02월 06일(화) 일일 프로젝트 진행 보고서 첨부파일로 송부 드립니다.</div>
    <br>
    <b>특이사항</b>
    <br>
    <div>- 해당 사항 없음.</div>
    <br>
    <div>감사합니다.</div>
    <div>정슬호 드림.</div>
  ",
  "title": "[SSAFY] 02월 06일(화) 구미 1반 실습코치 정슬호 일일 보고서 송부",
}
`;
